src/backend/distributed/commands/table.c

```diff
@@ -1390,3 +1450,52 @@ ErrorIfUnsupportedAlterAddConstraintStmt(AlterTableStmt *alterTableStatement)
 								 colocationId);
 	relation_close(relation, NoLock);
 }
+
+
+/*
+ * AlterTableSchemaStmtObjectAddress returns the ObjectAddress of the table that is the
+ * object of the AlterObjectSchemaStmt.
+ *
+ * This could be called both before or after it has been applied locally. It will look in
+ * the old schema first, if the table cannot be found in that schema it will look in the
+ * new schema. Errors if missing_ok is false and the table cannot be found in either of the
+ * schemas.
+ */
+ObjectAddress
+AlterTableSchemaStmtObjectAddress(Node *node, bool missing_ok)
+{
+	AlterObjectSchemaStmt *stmt = castNode(AlterObjectSchemaStmt, node);
+	Assert(stmt->objectType == OBJECT_TABLE);
+
+	const char *tableName = stmt->relation->relname;
+	Oid tableOid = InvalidOid;
+	if (stmt->relation->schemaname)
+	{
+		const char *schemaName = stmt->relation->schemaname;
+		Oid schemaOid = get_namespace_oid(schemaName, true);
```

Comment:
I think the second argument needs to be `false`.

Our code intercepts the call before it is applied locally. If the schemaname in the sql statement is non-existent I don't think we want to proceed with an `InvalidOid` for the schema.

A good rule of thumb is, we want to give the same error as postgres would give for this command.

Can you:
 a) make sure we have a test where we try to set the schema of a table with a qualified command for which the schema does not exist. eg. `ALTER TABLE non_existing_schema.my_table SET SCHEMA public`. Same for a table that does not exist.
b) compare the error message that vanilla postgres would give for these situations and see if we can easily provide the same errors.

Reply:
I'll change the trues to falses but then I wouldn't need to provide new error messages, because Postgres will already give the error.